cel/cel_test.go

```diff
@@ -70,6 +75,15 @@ func Example() {
 			Binary: func(lhs ref.Val, rhs ref.Val) ref.Val {
 				return types.String(
 					fmt.Sprintf("Hello %s! Nice to meet you, I'm %s.\n", rhs, lhs))
+			}},
+		&functions.Overload{
+			Operator: "shake_hands",
+			Binary: func(lhs ref.Val, rhs ref.Val) ref.Val {
+				// It is safe to skip ok with assertion if expression was type checked.
```

Comment:
In theory, we should still be type-testing the inputs. It's safer to use the values without type testing if you use the checker, but not exactly safe since you could get unexpected input as evidenced by the bugs you filed. Thanks for those reports, btw.

Reply:
I think when #210 is solved then it could be safe to use following approach. 
If you are not fine with it, I can use check is assertion was fine in examples however that was my idea how this could be simplified.

Reply:
The safety level really depends on how much you trust the input you're sending to the CEL expression. If you're always validating the input types and control the expressions, then type-testing isn't needed. If either the input or the expression leverages dynamic data (e.g. JSON, maps) which could be user provided then a type assertion of the arguments is strongly suggested.